### PR TITLE
Loading system proxy values by default

### PR DESCRIPTION
**Pull request by atais**
Like I have suggested in #10 I believe loading of System proxy values should be added.

I have created a simple code to handle that action.

----
Just a quick note:

I have found that generally, proxy topic is a bit broader: [link](https://stackoverflow.com/a/120802/1549135)

> Also don't forget the http.nonProxyHosts property! 

So this would mean some extra logic in the lib. However, I did not dive deeper into this as I did not find it necessary. Just mentioning fyi. 

Thanks!

### CODE REVIEW

File: core/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala [hunk 1]
```diff
@@ -37,12 +40,38 @@ object SttpBackendOptions {
...
-  val Default: SttpBackendOptions = SttpBackendOptions(30.seconds, None)
+  private val Empty: SttpBackendOptions =
+    SttpBackendOptions(30.seconds, None)
+
+  val Default: SttpBackendOptions =
+    Empty.copy(proxy = loadSystemProxy)
 
   def connectionTimeout(ct: FiniteDuration): SttpBackendOptions =
-    Default.connectionTimeout(ct)
+    Empty.connectionTimeout(ct)
```
Comment (adamw) [inline core/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala]: I'm wondering, wouldn't the end-user expect `SttpBackendOptions.connectionTimeout` to change only the connection timeout comparing to the defaults? In other words, shouldn't we modify `Default` here?

**Comment by adamw** [comment]
Ah and another thing - could you add yourself to `credits.rst`? :)

File: core/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala (hunk 1 above)
Comment (atais) [reply core/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala]: Wow. Did not think about it, but you are right. Fixed.

**Comment by adamw** [comment]
Released in 1.0.2 - thanks! :)

**Pull request merged**